add dovetail blog post (#1418)
carpentries/carpentries.org: _posts/2022/06/2022-06-08-dovtail-002.md

---
layout: page
authors: ["Zhian N. Kamvar"]
teaser: "Working through chopportunities"

> title: "The Dovetail: Updates from The Carpentries Workbench"

Author: ```suggestion
title: "The Dovetail #2: Updates from The Carpentries Workbench"
```

date: 2022-06-08
time: "09:00:00"
tags: ["Lesson Infrastructure", "Community", "Carpentries Workbench", "Beta", "Dovetail"]
---

> This is the second post in a series of posts that we are calling "The Dovetail."

Reviewer: ```suggestion
This is the second post in a series that we are calling "The Dovetail."
```

> In this series, we aim to keep members of The Carpentries community abreast on

Reviewer: ```suggestion
In this series, we aim to keep members of The Carpentries community abreast of
```

   - new method `$summary()` for lessons and episodes tabulates the different markdown elements.
   - Lessons can now analyze built files with the `$load_built()` method.
   - Bug fixes for showing `$error` and `$output` code blocks

To update your local Workbench installation, open R and use the following code:

```r
# Enable repository from carpentries
options(repos = c(
  ropensci = 'https://carpentries.r-universe.dev',
  CRAN = 'https://cloud.r-project.org'))
# Download and install sandpaper in R
install.packages(c('tinkr', 'pegboard', 'sandpaper', 'varnish'))
```

## Upcoming and Current Lessons in Workbench Beta

Schedule for the upcoming lessons coming soon! There are a few bottlenecks in
order to start beta testing for The Workbench:

1. We are still working on the correct workflow for uploading our lessons to AWS
   with the correct permissions so that we can dual deploy lessons for the
   stable and beta versions.
1. Our [lesson transition
   toolchain](https://github.com/data-lessons/lesson-transition#readme) is
   being modified to safely update lessons for entry into the workbench beta
   phase. It can enter lessons into pre-beta, but because the beta phase is a

>    destructive act, great care must be taken.

Reviewer: Can you link to somewhere community members can read more about the beta being "a destructive act"? I am not completely clear on what this means and the wording may be a little alarming for some readers!

Author: I've reworded it to say "irreversible" and link to the documentaiton

 - [Sarah Brown](https://github.com/brownsarahm/), Maintainer
 - [Joel Nitta](https://github.com/joel.nitta/), Translator
 - [Juan Fung](https://github.com/juanfung/), Maintainer
 - [Jannetta Steyn](https://github.com/jsteyn/), Lesson Developer (in The Carpentries Incubator)
 - [Michael Joseph](https://github.com/josephmje/), Lesson Developer (in The Carpentries Incubator)
 - [Sarah Stevens](https://github.com/sstevens2/), Lesson Developer (in The Carpentries Incubator)
 - [Kozo Nishida](https://github.com/kozo2/), Lesson Translation (to Japanese)
 - François Michonneau, Maintainer
 - [Jamie Jamison](https://github.com/jmjamison/), Maintainer
 - Jennifer Stubbs (she/her), Instructor
 - [Drake Asberry](https://github.com/drakeasberry/), Maintainer

## Updates from Community About Working in Workbench Beta

None yet! Stay tuned!

## Tips and Tricks for Using The Workbench

The [syntax for alt text in Workbench lessons](https://carpentries.github.io/workbench/transition-guide.html#figures)
is a little different than the syntax you may be used to:

```markdown
![caption](image.png){alt='alt text'}
```

The reason for this is because the markdown inside of `[]` can actually be

> rendered as markdown, so it makes more sense to use that as the figure caption.

Reviewer: Not sure what the correct wording should be, but "the markdown inside of `[]` can actually be rendered as markdown" is confusing to me.

Author: I've updated the wording to use "text" and "square braces" so I don't repeat "markdown" twice

> The alt text first and formost is text that needs to be accessible, so

Reviewer: ```suggestion
The alt text first and foremost is text that needs to be accessible, so
```

formatting is not as important.

Of course, your alt text is _rarely_ going to fit within a single line, so
instead of writing it as one long line, you can wrap the lines of text so that it
fits within the page. An example:

```markdown

> ![The author and his handsome son](fig/2022-06-08-author-cat.jpg){alt='A vertical
> selfie photograph of Zhian (a Middle-eastern man) with thick rimmed glasses and
> short dark hair wearing headphones. His grey cat is sitting in his lap, looking
> content as his head is being scratched.'}
> ```
> 
> 
> <figure style="text-align: center">
>   <img src="{{site.urlimg }}/blog/2022/06/2022-06-08-author-cat.jpg" 
>    alt="A vertical selfie photograph of Zhian (a Middle-eastern man) with thick rimmed glasses and short dark hair wearing headphones. His grey cat is sitting in his lap, looking content as his head is being scratched."/>
>   <figcaption>
>   The author and his handsome son
>   </figcaption>
> </figure>

Author: ```suggestion
![The author and his handsome son](fig/2022-06-08-author-cat.jpg){alt='A vertical
selfie photograph of Zhian (a Persian man) with thick-rimmed glasses and
short dark hair wearing headphones. His grey cat is sitting in his lap, looking
content as his head is being scratched.'}
```

<figure style="text-align: center">
  <img src="{{site.urlimg }}/blog/2022/06/2022-06-08-author-cat.jpg" 
   alt="A vertical selfie photograph of Zhian (a Persian man) with thick-rimmed glasses and short dark hair wearing headphones. His grey cat is sitting in his lap, looking content as his head is being scratched."/>
  <figcaption>
  The author and his handsome son
  </figcaption>
</figure>
```
